### PR TITLE
Revert "Revert "Create service agent on post""

### PR DESCRIPTION
**Pull request by kilbergr**
Reverts transcom/mymove#914
Redoes https://github.com/transcom/mymove/tree/rek_post-service-agent (will wait on @stangah to land his before I land--or at least talk to him about it)

### CODE REVIEW

File: pkg/handlers/service_agents_public.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	"github.com/transcom/mymove/pkg/gen/apimessages"
+	publicserviceagentop "github.com/transcom/mymove/pkg/gen/restapi/apioperations/service_agents"
+	"github.com/transcom/mymove/pkg/models"
+)
+
+/*
+ * ------------------------------------------
+ * The code below is for the PUBLIC REST API.
+ * ------------------------------------------
+ */
```
Comment (chrisgilmerproj) [inline pkg/handlers/service_agents_public.go]: You don't actually need these comments now that we're in `*_public.go` files anyway.

File: pkg/handlers/service_agents_public.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	payload := params.ServiceAgent
+
+	serviceAgentRole := models.Role(payload.Role)
+	newServiceAgent, verrs, err := models.CreateServiceAgent(
+		h.db,
+		shipment.ID,
+		serviceAgentRole,
+		*payload.PointOfContact,
```
[the remark this answers is not in the document]
Comment (kilbergr) [reply pkg/handlers/service_agents_public.go]: I'm changing it so it's unwrapped in the payload--obviously just causing confusion to do this way.

[the remark this answers is not in the document]
Comment (kilbergr) [reply pkg/handlers/service_agents_public.go]: Changing it so this is no longer confusing.

File: pkg/handlers/service_agents_public.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	// TODO (rebecca): Find a way to check Shipment belongs to TSP without 2 queries
+	tspUser, err := models.FetchTspUserByID(h.db, session.TspUserID)
+	if err != nil {
+		h.logger.Error("DB Query", zap.Error(err))
+		return publicserviceagentop.NewCreateServiceAgentForbidden()
+	}
+
+	shipment, err := models.FetchShipmentByTSP(h.db, tspUser.TransportationServiceProviderID, shipmentID)
```
[the remark this answers is not in the document]
Comment (kilbergr) [reply pkg/handlers/service_agents_public.go]: There's a todo above FetchTspUserByID

**Pull request merged**